frontend/cmd/cmd.go

```
@@ -0,0 +1,137 @@
+	rootCmd.Flags().BoolVar(&opts.insecure, "insecure", false, "Skip validating TLS for clusters-service.")
+
+	rootCmd.MarkFlagsMutuallyExclusive("use-cache", "database-name")
+	rootCmd.MarkFlagsRequiredTogether("database-name", "database-url")
+
+	return rootCmd
+}
+
+func (opts *FrontendOpts) Run() error {
+	version := "unknown"
+	if info, ok := debug.ReadBuildInfo(); ok {
+		for _, setting := range info.Settings {
+			if setting.Key == "vcs.revision" {
+				version = setting.Value
+				break
+			}
+		}
+	}
+
+	logger := config.DefaultLogger()
+	logger.Info(fmt.Sprintf("%s (%s) started", frontend.ProgramName, version))
+
+	// Init prometheus emitter
+	prometheusEmitter := frontend.NewPrometheusEmitter()
+
+	// Configure database configuration and client
+	var dbClient database.DBClient
+	if opts.useCache {
+		dbClient = database.NewCache()
+	} else {
+		var err error
+
+		dbConfig := database.NewDatabaseConfig(opts.databaseName, opts.databaseURL)
+		dbClient, err = database.NewDatabaseClient(dbConfig)
+		if err != nil {
+			return fmt.Errorf("creating the database client failed: %v", err)
+		}
+	}
+
+	listener, err := net.Listen("tcp4", fmt.Sprintf(":%d", opts.port))
+	if err != nil {
+		return err
+	}
+
+	logger.Info(fmt.Sprintf("Application running in region: %s", opts.region))
+
+	// Initialize Clusters Service Client
+	conn, err := sdk.NewUnauthenticatedConnectionBuilder().
+		URL(opts.clustersServiceURL).
+		Insecure(opts.insecure).
+		Build()
+	if err != nil {
+		return err
+	}
+
+	f := frontend.NewFrontend(logger, listener, prometheusEmitter, dbClient, opts.region, conn)
+
+	// Verify the Async DB is available and accessible
+	logger.Info("Testing DB Access")
+	result, err := f.DbClient.DBConnectionTest(context.Background())
+	if err != nil {
+		logger.Error(fmt.Sprintf("Database test failed to fetch properties: %v", err))
+	} else {
+		logger.Info(fmt.Sprintf("Database check completed - %s", result))
+	}
```

Quick question: should we return an error (and avoid the start of the app) instead of just logging the message and continuing?

---

Maybe in the near future after this PR merges to introduce a local cache as an alternative to CosmosDB, not for now though to avoid blocking development/testing.